website/docs/flink_configuration.md

```diff
@@ -3,115 +3,177 @@ title: Flink Setup
+
+All Hudi-specific parallelism and memory configurations depend on your Flink job resources.
+
+When using Flink, you can set some global configurations in `$FLINK_HOME/conf/flink-conf.yaml`.
+
+### Parallelism​
+
+If your system has a lot of data to ingest, increasing the parallelism can improve throughput significantly. Hudi supplies flexible config options for specific operators, but at a high level, a default global parallelism can reduce the complexity of manual configuration. Try the default configuration and adjust as necessary. 
+
+| Property Name | Default  | Description | Scope | Since Version                          |
+|----------------|--------|----------|---------------|--------------------------------------|
+| `taskmanager.numberOfTaskSlots` | 1 | The is the number of parallel operators or user function instances that a single TaskManager can run. We recommend setting this value > 4, and the actual value needs to be set according to the amount of data | n/a | 0.9.0 |
+| `parallelism.default` | 1 | The is the default parallelism used when no parallelism is specified anywhere (default: 1). For example, if the value of [`write.bucket_assign.tasks`](#parallelism-1) is not set, this value will be used | n/a | 0.9.0.|
+
+### Memory​
+The `JobManager` and `TaskManager` memory configuration is very important for a Flink job to work smoothly. Below, we'll describe these configurations. 
+
+#### JobManager
+The JobManager handles all the instants coordination. It keeps an in-memory fs view for all the file handles on the filesystem within its embedded timeline server. We need to ensure enough memory is allocated to avoid OOM errors. The configs below allow you to allocate the necessary memory. 
+
+| Property Name | Default  | Description | Scope | Since Version                          |
+|----------------|--------|----------|---------------|--------------------------------------|
+| `jobmanager.memory.process.size` | -- |This is the total process memory size for the JobManager. This includes all the memory that a JobManager JVM process consumes: Total Flink Memory, JVM Metaspace, and JVM Overhead | n/a | 0.9.0
+
+
+#### TaskManager
+The TaskManager is a container for the writing and table service tasks. For regular Parquet file flushing, we need to allocate enough memory to read and write files. At the same time, there must be enough resources for  MOR table compaction because it’s memory intensive: we need to read and merge all the log files into an output Parquet file. Below are the configs you can set for the TaskManager to allocate enough memory for these services. 
```

Comment:
```suggestion
The TaskManager is a container for the writing and table service tasks. For regular Parquet file flushing, we need to allocate enough memory to read and write files. At the same time, there must be enough resources for  MOR table compaction because it’s memory intensive: we need to read and merge all the log files into an output base file. Below are the configs you can set for the TaskManager to allocate enough memory for these services. 
```

Reply:
lets move away from parquet to base

Reply:
updated

Reply:
updated